File: z-clients/java/src/main/java/com/svix/coyote/apis/AdminAuthToken.java
// this file is @generated
package com.svix.coyote.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.coyote.ApiException;
import com.svix.coyote.HttpClient;
import com.svix.coyote.Utils;
import java.io.IOException;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;
import java.util.Set;
import okhttp3.Headers;
import okhttp3.HttpUrl;
import com.svix.coyote.models.AdminAuthTokenCreateIn;
import com.svix.coyote.models.AdminAuthTokenCreateOut;
import com.svix.coyote.models.AdminAuthTokenDeleteIn;
import com.svix.coyote.models.AdminAuthTokenDeleteOut;
import com.svix.coyote.models.AdminAuthTokenExpireIn;
import com.svix.coyote.models.AdminAuthTokenExpireOut;
import com.svix.coyote.models.AdminAuthTokenListIn;
import com.svix.coyote.models.AdminAuthTokenRotateIn;
import com.svix.coyote.models.AdminAuthTokenRotateOut;
import com.svix.coyote.models.AdminAuthTokenUpdateIn;
import com.svix.coyote.models.AdminAuthTokenUpdateOut;
import com.svix.coyote.models.AdminAuthTokenWhoamiIn;
import com.svix.coyote.models.AdminAuthTokenWhoamiOut;
import com.svix.coyote.models.ListResponseAdminAuthTokenOut;

public class AdminAuthToken {
    private final HttpClient client;

    public AdminAuthToken(HttpClient client) {
        this.client = client;
    }

    /** Create an auth token */
    public AdminAuthTokenCreateOut create(
        final AdminAuthTokenCreateIn adminAuthTokenCreateIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1.admin.auth-token.create");

        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            adminAuthTokenCreateIn,
            AdminAuthTokenCreateOut.class
        );
    }

    /** Expire an auth token */
    public AdminAuthTokenExpireOut expire(
        final AdminAuthTokenExpireIn adminAuthTokenExpireIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1.admin.auth-token.expire");

        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            adminAuthTokenExpireIn,
            AdminAuthTokenExpireOut.class
        );
    }

    /** Rotate an auth token, invalidating the old one and issuing a new secret */
    public AdminAuthTokenRotateOut rotate(
        final AdminAuthTokenRotateIn adminAuthTokenRotateIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1.admin.auth-token.rotate");

        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            adminAuthTokenRotateIn,
            AdminAuthTokenRotateOut.class
        );
    }

    /** Delete an auth token */
    public AdminAuthTokenDeleteOut delete(
        final AdminAuthTokenDeleteIn adminAuthTokenDeleteIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1.admin.auth-token.delete");

        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            adminAuthTokenDeleteIn,
            AdminAuthTokenDeleteOut.class
        );
    }

    /** List auth tokens for a given owner */
    public ListResponseAdminAuthTokenOut list(
        final AdminAuthTokenListIn adminAuthTokenListIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1.admin.auth-token.list");

        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            adminAuthTokenListIn,
            ListResponseAdminAuthTokenOut.class
        );
    }

    /** List auth tokens for a given owner */
    public ListResponseAdminAuthTokenOut list(
        
    ) throws IOException, ApiException {
        return this.list(
            new AdminAuthTokenListIn()
        );
    }

    /** Update an auth token's properties */
    public AdminAuthTokenUpdateOut update(
        final AdminAuthTokenUpdateIn adminAuthTokenUpdateIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1.admin.auth-token.update");

        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            adminAuthTokenUpdateIn,
            AdminAuthTokenUpdateOut.class
        );
    }

    /** Return the role of the currently authenticated token */
    public AdminAuthTokenWhoamiOut whoami(
        final AdminAuthTokenWhoamiIn adminAuthTokenWhoamiIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1.admin.auth-token.whoami");

        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            adminAuthTokenWhoamiIn,
            AdminAuthTokenWhoamiOut.class
        );
    }

    /** Return the role of the currently authenticated token */
    public AdminAuthTokenWhoamiOut whoami(
        
    ) throws IOException, ApiException {
        return this.whoami(
            new AdminAuthTokenWhoamiIn()
        );
    }
}

File: z-clients/javascript/src/apis/adminAuthToken.ts
// this file is @generated

import {
    type AdminAuthTokenCreateIn,
    AdminAuthTokenCreateInSerializer,
} from '../models/adminAuthTokenCreateIn';
import {
    type AdminAuthTokenCreateOut,
    AdminAuthTokenCreateOutSerializer,
} from '../models/adminAuthTokenCreateOut';
import {
    type AdminAuthTokenDeleteIn,
    AdminAuthTokenDeleteInSerializer,
} from '../models/adminAuthTokenDeleteIn';
import {
    type AdminAuthTokenDeleteOut,
    AdminAuthTokenDeleteOutSerializer,
} from '../models/adminAuthTokenDeleteOut';
import {
    type AdminAuthTokenExpireIn,
    AdminAuthTokenExpireInSerializer,
} from '../models/adminAuthTokenExpireIn';
import {
    type AdminAuthTokenExpireOut,
    AdminAuthTokenExpireOutSerializer,
} from '../models/adminAuthTokenExpireOut';
import {
    type AdminAuthTokenListIn,
    AdminAuthTokenListInSerializer,
} from '../models/adminAuthTokenListIn';
import {
    type AdminAuthTokenRotateIn,
    AdminAuthTokenRotateInSerializer,
} from '../models/adminAuthTokenRotateIn';
import {
    type AdminAuthTokenRotateOut,
    AdminAuthTokenRotateOutSerializer,
} from '../models/adminAuthTokenRotateOut';
import {
    type AdminAuthTokenUpdateIn,
    AdminAuthTokenUpdateInSerializer,
} from '../models/adminAuthTokenUpdateIn';
import {
    type AdminAuthTokenUpdateOut,
    AdminAuthTokenUpdateOutSerializer,
} from '../models/adminAuthTokenUpdateOut';
import {
    type AdminAuthTokenWhoamiIn,
    AdminAuthTokenWhoamiInSerializer,
} from '../models/adminAuthTokenWhoamiIn';
import {
    type AdminAuthTokenWhoamiOut,
    AdminAuthTokenWhoamiOutSerializer,
} from '../models/adminAuthTokenWhoamiOut';
import {
    type ListResponseAdminAuthTokenOut,
    ListResponseAdminAuthTokenOutSerializer,
} from '../models/listResponseAdminAuthTokenOut';
import { HttpMethod, DiomRequest, type DiomRequestContext } from "../request";

export class AdminAuthToken {
    public constructor(private readonly requestCtx: DiomRequestContext) {}

    /** Create an auth token */
    public create(
        adminAuthTokenCreateIn: AdminAuthTokenCreateIn,
    ): Promise<AdminAuthTokenCreateOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.admin.auth-token.create");

        request.setBody(
            AdminAuthTokenCreateInSerializer._toJsonObject(adminAuthTokenCreateIn)
        );
        
        return request.send(
            this.requestCtx,
            AdminAuthTokenCreateOutSerializer._fromJsonObject,
        );
    }/** Expire an auth token */
    public expire(
        adminAuthTokenExpireIn: AdminAuthTokenExpireIn,
    ): Promise<AdminAuthTokenExpireOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.admin.auth-token.expire");

        request.setBody(
            AdminAuthTokenExpireInSerializer._toJsonObject(adminAuthTokenExpireIn)
        );
        
        return request.send(
            this.requestCtx,
            AdminAuthTokenExpireOutSerializer._fromJsonObject,
        );
    }/** Rotate an auth token, invalidating the old one and issuing a new secret */
    public rotate(
        adminAuthTokenRotateIn: AdminAuthTokenRotateIn,
    ): Promise<AdminAuthTokenRotateOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.admin.auth-token.rotate");

        request.setBody(
            AdminAuthTokenRotateInSerializer._toJsonObject(adminAuthTokenRotateIn)
        );
        
        return request.send(
            this.requestCtx,
            AdminAuthTokenRotateOutSerializer._fromJsonObject,
        );
    }/** Delete an auth token */
    public delete(
        adminAuthTokenDeleteIn: AdminAuthTokenDeleteIn,
    ): Promise<AdminAuthTokenDeleteOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.admin.auth-token.delete");

        request.setBody(
            AdminAuthTokenDeleteInSerializer._toJsonObject(adminAuthTokenDeleteIn)
        );
        
        return request.send(
            this.requestCtx,
            AdminAuthTokenDeleteOutSerializer._fromJsonObject,
        );
    }/** List auth tokens for a given owner */
    public list(
        adminAuthTokenListIn: AdminAuthTokenListIn,
    ): Promise<ListResponseAdminAuthTokenOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.admin.auth-token.list");

        request.setBody(
            AdminAuthTokenListInSerializer._toJsonObject(adminAuthTokenListIn)
        );
        
        return request.send(
            this.requestCtx,
            ListResponseAdminAuthTokenOutSerializer._fromJsonObject,
        );
    }/** Update an auth token's properties */
    public update(
        adminAuthTokenUpdateIn: AdminAuthTokenUpdateIn,
    ): Promise<AdminAuthTokenUpdateOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.admin.auth-token.update");

        request.setBody(
            AdminAuthTokenUpdateInSerializer._toJsonObject(adminAuthTokenUpdateIn)
        );
        
        return request.send(
            this.requestCtx,
            AdminAuthTokenUpdateOutSerializer._fromJsonObject,
        );
    }/** Return the role of the currently authenticated token */
    public whoami(
        adminAuthTokenWhoamiIn: AdminAuthTokenWhoamiIn,
    ): Promise<AdminAuthTokenWhoamiOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.admin.auth-token.whoami");

        request.setBody(
            AdminAuthTokenWhoamiInSerializer._toJsonObject(adminAuthTokenWhoamiIn)
        );
        
        return request.send(
            this.requestCtx,
            AdminAuthTokenWhoamiOutSerializer._fromJsonObject,
        );
    }
}

